Modules/Product/Http/Services/ProductService.php

```diff
@@ -69,12 +76,12 @@ public function fetch($request): array
         }
         $products = $query->paginate(10);
         foreach ($products as $v) {
-            $productVariations = ProductVariation::where('product_id', $v->id)->where('status', '1')->get();
+            $productVariations = Product::find($v->id)->productVariation()->where('status', '1')->get();
             $productVariationsCount = $productVariations->count();
             if ($productVariationsCount > 0) {
-                $variantMinPrice = ProductVariation::where('product_id', $v->id)->min('price');
+                $variantMinPrice = Product::find($v->id)->productVariation()->min('price');
                 $price = $variantMinPrice . '+';
-                $variantStock = ProductVariation::where('product_id', $v->id)->sum('price');
+                $variantStock = Product::find($v->id)->productVariation()->sum('price');
                 $availability = $variantStock > 0 ? 'in stock' : 'out of stock';
             } else {
```

Comment:
As i told you before, do not make a query inside foreach loop. In this code you're requesting multiple queries inside the loop which will effect the API performance when there will be large data.

Points to be improved:

1. First rename the $v to $product, use proper names that are understandable.
2. get the relation using the eager loading in product query like Product::with('productVariation')..... and inside the loop where you need to get further data from product variations, use like this $product->productVariation()->further query...
this way all your queries inside the loop will be eliminated.